solvers/relay_constellation/umcf_srr_contact_plan/src/dynamic_graph.py

```
@@ -0,0 +1,135 @@
+            node_a=edge.node_b,
+            node_b=edge.node_a,
+            distance_m=edge.distance_m,
+            capacity=edge.capacity,
+        )
+        self.adjacency.setdefault(edge.node_b, []).append(reverse)
+
+
+def build_sample_graphs(
+    case: Case,
+    satellites: dict[str, Satellite],
+    positions_ecef: dict[str, np.ndarray] | None = None,
```

Ruff (0.15.11)

[error] Undefined name `np`

(F821)